dockstore-webservice/src/main/java/io/dockstore/webservice/resources/EntryResource.java

```diff
@@ -547,4 +548,31 @@ public Entry getAndCheckResource(User user, Long id) {
     public Entry getAndCheckResourceByAlias(String alias) {
         throw new UnsupportedOperationException("Use the TRS API for tools and workflows");
     }
+
+    @Override
+    public void checkCanRead(User user, Entry workflow) {
+        try {
+            checkUser(user, workflow);
+        } catch (CustomWebApplicationException ex) {
+            checkCanReadAcrossEntryTypes(user, workflow, permissionsInterface, ex);
+        }
+    }
```

Comment:
Could we override `checkUser` and put all this logic into it, instead?  That way, the fix would propagate with fewer changes, and we'd have fewer subtle variants of the check methods.

Reply:
Oh wait, nevermind.  Looking at the code some more, I think I get it.  `checkUser` checks if a user "owns" the entry, and `checkCanRead` checks if the user can read it.

Reply:
Upon further review, that's not true, either.

Reply:
Ostensibly, the comments in `AuthenticatedResourceInterface.java` define when to use each of the `check` methods.  However, I think some (most?) of our code has diverged from those guidelines.

Reply:
More observations regarding this matter:
1. There are three different methods in `AuthenticatedResourceInterface` with names/signatures that imply that they might check if a user can read an entry: `checkCanRead`, `checkUserCanRead`, and `checkOptionalAuthRead`.
2. IMHO, if they do different things, we need to name them more clearly, and if not, eliminate the dups.
3. A good implementation would check if the entry was published, check if the user was admin, check if the user owned the entry, and if none of the above, delegate to another method that checked the user/entry against the optional authentication methods defined by implementing classes.

Reply:
FWIW, I was a little hesitant about making "this level" of changes in a hotfix. 
I think a clean refactor is definitely develop territory (but also a very good idea!)

Reply:
Creating https://ucsc-cgl.atlassian.net/browse/SEAB-4403 for this discussion